### PR TITLE
Graceful failure with good error message when no solidity files found !

### LINKED ISSUE #543: Tests for graceful exit condition on empty contexts .
https://github.com/Cyfrin/aderyn/pull/499

### PR DESCRIPTION
**Pull request by TilakMaddy**
fix #543

### CODE REVIEW

File: aderyn_driver/src/foundry_compiler_helpers.rs [hunk 1]
```diff
@@ -17,6 +18,10 @@ pub fn get_compiler_input(root: &Path) -> CompilerInput {
         .into_iter()
         .filter(|c| c.language == *"Solidity")
         .collect::<Vec<_>>();
+    if solidity_files.is_empty() {
+        eprintln!("No solidity files found in {}!", root.to_string_lossy());
+        exit(1);
+    }
     let solidity_files = &solidity_files[0]; // No Yul Support as of now
```
Comment (TilakMaddy) [inline aderyn_driver/src/foundry_compiler_helpers.rs:25]: This line panics if the above check isn't made

File: aderyn_driver/src/driver.rs [hunk 2]
```diff
@@ -115,6 +115,11 @@ fn make_context(args: &Args) -> WorkspaceContextWrapper {
     let mut contexts: Vec<WorkspaceContext> =
         process_auto::with_project_root_at(&root_path, &src, &exclude, &remappings, &include);
 
+    if contexts.iter().all(|c| c.src_filepaths.is_empty()) {
+        eprintln!("No solidity files found in given scope!");
```
Comment (alexroan) [inline aderyn_driver/src/driver.rs:119]: Can you introduce a test for this please @TilakMaddy ?

File: aderyn_driver/src/foundry_compiler_helpers.rs [hunk 3]
```diff
@@ -17,6 +18,10 @@ pub fn get_compiler_input(root: &Path) -> CompilerInput {
         .into_iter()
         .filter(|c| c.language == *"Solidity")
         .collect::<Vec<_>>();
+    if solidity_files.is_empty() {
```
Comment (alexroan) [inline aderyn_driver/src/foundry_compiler_helpers.rs:21]: Test here too please

**Review by alexroan — CHANGES_REQUESTED** [review]

**Comment by alexroan** [comment]
@TilakMaddy can you please add those tests in here? :)

**Comment by TilakMaddy** [comment]
Sorry, I forgot about this. Will add tests shortly : )

File: aderyn_driver/src/foundry_compiler_helpers.rs (hunk 3 above)
Comment (TilakMaddy) [reply aderyn_driver/src/foundry_compiler_helpers.rs:21]: This is to check for safe array index access in the next line ` let solidity_files = &solidity_files[0]; `

We must ensure vector is not empty if we want to access index 0 of the vector : )

File: aderyn_driver/src/driver.rs (hunk 2 above)
Comment (TilakMaddy) [reply aderyn_driver/src/driver.rs:128]: What might we test for in this case ?

Say we form a contexts vector from a directory of solidity files but with no solidity files matching the given scope.

Like, `let mut contexts: Vec<WorkspaceContext> =
        process_auto::with_project_root_at(&root_path, &src, &exclude, &remappings, &include);`

Then test for `contexts.iter().all(|c| c.src_filepaths.is_empty()) == true`  is alright?

Comment (TilakMaddy) [reply aderyn_driver/src/driver.rs:128]: DONE ✅

File: aderyn_driver/src/foundry_compiler_helpers.rs (hunk 3 above)
Comment (TilakMaddy) [reply aderyn_driver/src/foundry_compiler_helpers.rs:21]: Idea - 💡 Let me write a test for this by having a directory with no solidity files at all whatsoever.

**Review by alexroan — APPROVED** [review]

**Pull request merged**